libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/InMemoryDataStore.java

```diff
@@ -0,0 +1,205 @@
+package uk.gov.di.ipv.core.library.persistence;
+
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbSecondaryPartitionKey;
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbSortKey;
+import uk.gov.di.ipv.core.library.config.ConfigurationVariable;
+import uk.gov.di.ipv.core.library.exceptions.ItemAlreadyExistsException;
+import uk.gov.di.ipv.core.library.persistence.item.PersistenceItem;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class InMemoryDataStore<T extends PersistenceItem> implements DataStore<T> {
+    private static final String KEY_SEPARATOR = "/";
+    private static final ConcurrentMap<String, Map<String, ?>> TABLES = new ConcurrentHashMap<>();
+
+    private final ConcurrentMap<String, T> records;
```

Comment:
Why ConcurrentMap? Is there a particular need for thread safety here?

Reply:
This is effectively the dynamodb 'table', which is stored statically in `TABLES` (but stored as an instance variable as well, for convenience) - this might be accessed by multiple requests in parallel, so I think we need thread safety.

Reply:
OK, because we're departing from running in the lambda runtime. If we were to run the tests suites against local we could end up with concurrent requests.